chingu-voyage6/Bears-Team-07
# Feature/puffs images

First try at implementing uploading puff images.

# Breaking Changes:
- The backend, no longer forces json content type, you now have to explicitly set it for POST/PUT and probably DELETE, any GETs works fine without it.  This is needed as with upload you it will send a different content type.

# Run the backend
- Create a new branch based from development or create a new branch based from your topic branch (the current branch you working on) then merge this branch/pull request to it.
    - `git checkout development -b test-puffs-images` or `git checkout topic-branch -b test-puffs-images`
    - `git fetch --prune && git merge feature/puffs-images` or `git fetch --prune && git merge origin/feature/puffs-images`
    - cd server
    - yarn install 
    - yarn run dev

# To test using Postman:
1 - Create a POST (Create)
2 - Set URL to: http://localhost:3000/api/v1/puffs/u
3 - On headers, set contenttype to `application/x-www-form-urlencoded`
4 - On body, use `form-data`
4.1 - Key: `upload` as File #Key must be `upload` the backend expects that.
4.2 - Value: Choose a file to upload
5 - Create a PUT (Update)
6 - Set URL to: http://localhost:3000/api/v1/puffs/u/:id
6.1 -  `:id` is similar to `5b3e8917dc0d23dee6f852be` get it from `_id` from the puff database - The puff to upload an image to.
6.2 - As you can see puff needs to be created first then you can upload the image.
6.3 - Repeat 3 and 4 above.
